src/verifier.rs

```diff
@@ -223,23 +306,52 @@ fn verify_all_ipas<C: HaloCurve>(
     // Then, we reduce the above opening set reductions to a single value.
     let reduced_opening = reduce_with_powers(&opening_set_reductions, v);
 
-    let u_curve = C::convert(u_scaling) * circuit.u.to_projective();
-
-    verify_ipa::<C>(proof, c_reduction, reduced_opening, point, ipa_challenges, u_curve, circuit.pedersen_h,proof.halo_g, schnorr_challenge, proof.schnorr_proof)
+    let u_curve = C::convert(u_scaling) * u_curve.to_projective();
```

Comment:
I'd prefer if the variable names differed -- maybe call the new one u_prime or u_scaled?